### ISSUE TITLE
Accessibility

### ISSUE THREAD

**Issue by sepulzera** [ownrecipes/OwnRecipes#99]
* [x] The html lang attribute does not reflect the selected language.
* [x] Icon Buttons have no label.
* [x] Link icon buttons have two tab-indice.
* [x] View recipe: servings input field should have a unit ("servings").
* [x] View recipe: Update servings button reads "servings", that does not convey it's action clearly.
* [x] View recipe: caption id may not contain whitespace.
* [x] View recipe > Ratings: Delete button has no alt textl.
* [x] View recipe > Add to grocery list: checkboxes should read their items.
* [x] View recipe > Add to grocery list: modal close button lacks translation.
* [x] View recipe > Add to grocery list: Add list is not correctly connected with its label.
* [x] View recipe > Add to grocery list: Add list button aria-label lacks translation.
* [x] New rating: Stars have no label.
* [x] ~Header navigation should be marked as list (ol, li).~ *1
* [x] Header navigation active element should be marked as "aria-current".
* [x] Search: filter section should be marked as article or aside.
* [x] Search: filter categories show/hide buttons have no visible focus.
* [x] Search: active filters not marked.
* [x] Search: reset filter buttons should only be visible when they have an effect.
* [x] Search: Clear text input button has no alt text.
* [x] Search: pagination should be wrapped with "nav".
* [x] Search: pagination previous and next buttons have no alt text.
* [x] Search: it is tedious to skip the filters and navigate to a search result via keyboard (f. e. there are no headings).
* [x] Select dropdown: the text contrast for active item (is disabled) is too low.
* [x] input of type text may not have min/max attributes.
* [x] h3 not allowed as child of element span (see card-titles for recipe results).
* [x] Windows High contrast mode:
    * [x] active list items have no visual indicator
    * [x] chips have no border (view recipe)
    * [x] footer has no border

**Comment by sepulzera** [comment]
1) Decided to not tinker with the header. There is too much JS magic going on with collapse and stuff. Fixing this with [proper ul+li](https://react-bootstrap.github.io/components/navs/#base-nav) would be quite a huge change.
